### DISCUSSION TITLE
How can I read two fields on a contract at once?

### DISCUSSION THREAD

**Discussion by cilphex** [wevm/wagmi#1548] [Q&A]
Say there's a contract with two fields:

```
pragma solidity ^0.8.17;

contract TestContract {
    string public text1 = "uno";
    string public text2 = "dos";
}
```

Is it possible to read them both in one call using `wagmi/core`?

I can read one field this way:

```
contractText = await readContract({
    address,
    abi,
    functionName: "text1",
});
```

But I don't know how to read both together. Calling `readContract` twice in a row seems inefficient.

**Comment by tmm** [comment]
Check out [`readContracts`](https://wagmi.sh/core/actions/readContracts).
